cocos2d/audio/CCAudio.js

```diff
@@ -44,8 +44,11 @@ let Audio = function (src) {
 
     const self = this;
     this._onended = function () {
-        self._state = Audio.State.STOPPED;
-        self.emit('ended');
+        if(self.paused){
```

Comment:
If self.pused is true, it should need to be changed to false ?  Why it can be true, we may call stop firstly.